dev/io.openliberty.checkpoint_fat/fat/src/io/openliberty/checkpoint/fat/FacesTest.java

```diff
@@ -0,0 +1,143 @@
+                             "/META-INF/microprofile-config.properties")
+                        .add(new FileAsset(new File("test-applications/eventMgr/frontendUI/resources/META-INF/MANIFEST.MF")),
+                             "/META-INF/MANIFEST.MF")
+                        .add(new FileAsset(new File("test-applications/eventMgr/frontendUI/resources/navBar/leftNav.xhtml")),
+                             "/navBar/leftNav.xhtml")
+                        .add(new FileAsset(new File("test-applications/eventMgr/frontendUI/resources/WEB-INF/faces-config.xml")),
+                             "/WEB-INF/faces-config.xml")
+                        .add(new FileAsset(new File("test-applications/eventMgr/frontendUI/resources/WEB-INF/web.xml")),
+                             "/WEB-INF/web.xml");
+        ShrinkHelper.exportAppToServer(frontendUI, frontendUIWar);
+
+        backendServices = LibertyServerFactory.getLibertyServer(BACKEND_SERVER_NAME);
+        WebArchive backendServicesWar = ShrinkWrap.create(WebArchive.class, "backendServices.war")
+                        .addPackages(true, "io.openliberty.guides.event.dao")
+                        .addPackages(true, "io.openliberty.guides.event.models")
+                        .addPackages(true, "io.openliberty.guides.event.resources")
+                        .add(new FileAsset(new File("test-applications/eventMgr/backendServices/resources/META-INF/persistence.xml")),
+                             "/WEB-INF/classes/META-INF/persistence.xml")
+                        .add(new FileAsset(new File("test-applications/eventMgr/backendServices/resources/META-INF/MANIFEST.MF")),
+                             "/META-INF/MANIFEST.MF");
+        ShrinkHelper.exportAppToServer(backendServices, backendServicesWar);
+        // Skip this. Use derby provided in ${shared.resources.dir}
+        //backendServices.copyFileToLibertyServerRoot("publish/shared/resources/derby", "derbyLib", "derby.jar");
+    }
+
+    @Before
+    public void setUp() throws Exception {
+        frontendUI.setCheckpoint(CheckpointPhase.APPLICATIONS, true,
+                                 server -> {
+                                     assertNotNull("'SRVE0169I: Loading Web Module: frontendUI' message not found in log before restore",
+                                                   server.waitForStringInLogUsingMark("SRVE0169I: Loading Web Module: frontendUI", 0));
+                                     assertNotNull("'CWWKZ0001I: Application frontendUI started' message not found in log.",
+                                                   server.waitForStringInLogUsingMark("CWWKZ0001I: Application frontendUI started", 0));
+                                 });
+        frontendUI.startServer(getTestMethodNameOnly(testName) + ".log");
+    }
+
+    @After
+    public void tearDown() throws Exception {
+        frontendUI.stopServer();
```

Comment:
We need to ensure that both servers stop.  If stopping the `frontendUI` server fails then we will not stop the `backendServices` server.  Perhaps you can have two `@After` methods, one to stop each server?

Reply:
I'll fix this by keeping things consistent: stop `backendServices` in the test that starts it; stop `frontendUI` in the `@After tearDown`